Makefile

```diff
@@ -40,6 +40,9 @@ BUNDLE_GEN_FLAGS ?= -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 
 VERIFY_TLS ?= true
 
+# GOWORK
+GOWORK ?= off
+
```

Comment:
Why don't we use the Make `export` construct so that it is present in all the executions?

That would save us from adding GOWORK everywhere in the `Makefile`

```
export GOWORK := $GOWORK
```

Reply:
> Why don't we use the Make `export` construct so that it is present in all the executions?
> 
> That would save us from adding GOWORK everywhere in the `Makefile`
> 
> ```
> export GOWORK := $(GOWORK)
> ```

Not sure I'm properly following you here, but as per [1] (if it lands) I can remove all the `GOWORK=$(GOWORK)` occurrences in the `Makefile` and just have `GOWORK ?= off` at the top of the file as you previously suggested.

Then, if you run something like:
```
GOWORK=on make govet
```

you end up seeing something like [2], but if you run:

```
GOWORK=off make govet 
``` 
(or just `make govet`) you'll see something like [3].

IMO this would simplify a lot the `Makefile` and we don't have to deal w/ that variable unless we would like to set it to `on`.

@Akrog does this make sense?


[2] https://paste.opendev.org/show/bLGg1Dx7D4ueNmqIT5kQ/
[3] https://paste.opendev.org/show/b48LCNwZFoOk3DdPKraI/

Reply:
If you just add:
```
GOWORK ?= off
export GOWORK := $(GOWORK)
```
Then you don't need to add `GOWORK=$(GOWORK)` anywhere in this file.
And then we you want to use the go workspace you can do:
```
GOWORK= make build
```
And it will use the current go workspace.

Reply:
My only point here is that we can avoid `export GOWORK := $(GOWORK)` if [1] lands.
However, there isn't much change in how `govet.sh` and the other scripts are called, I guess the only thing we should take care is documenting you have to `GOWORK= make build` (before this change you didn't have to).
If we combine this change with [1] we can't just avoid the `export` directive (which defaults to `off` because of the CI execution) and keep the user experience as it is (and in line with the other operators).

[1] https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/pull/56

Reply:
I think everything in `Makefile` should not use the go workspace by default, That way build and run also ignores it unless explicitly requested with the `GOWORK=`.

I'm thinking about leftover `go.work` replaces after testing a PR and then forgetting about it and doing `make build` and getting all kinds of issues.

I believe it is less likely to forget adding the `GOWORK=` when we are working with dependencies than the other way around.

The [documentation PR already includes the `GOWORK=`](https://github.com/openstack-k8s-operators/cinder-operator/pull/77/commits/27f69cf3044bddbdb464acecef7c5b67d086a050) when using external dependencies.

Reply:
> I think everything in `Makefile` should not use the go workspace by default, That way build and run also ignores it unless explicitly requested with the `GOWORK=`.
> 
> I'm thinking about leftover `go.work` replaces after testing a PR and then forgetting about it and doing `make build` and getting all kinds of issues.
> 
> I believe it is less likely to forget adding the `GOWORK=` when we are working with dependencies than the other way around.

Agreed, tbh I didn't have a strong opinion but as you mentioned, it's good making sure that people understand how to explicitly turn it on instead of messing with potential `go.work` changes (that would end up "working on the laptop").
Being that said, I added the `export` directive in the last iteration.
> 
> The [documentation PR already includes the `GOWORK=`](https://github.com/openstack-k8s-operators/cinder-operator/pull/77/commits/27f69cf3044bddbdb464acecef7c5b67d086a050) when using external dependencies.

Thank you, sounds good!